### PR TITLE
MueLu: fix issue #3035

### PR DESCRIPTION
**Pull request by jhux2**
Auto-PR for SHA ecfc597

---
This change is [Reviewable](https://reviewable.io/reviews/trilinos/trilinos/3150)

### CODE REVIEW

**Comment by mhoemmen** [comment]
Same deal -- different number of partitions than expected in the MueLu Epetra unit test.

File: packages/muelu/research/mmayr/composite_to_regions/src/CMakeLists.txt [hunk 1]
```diff
@@ -6,12 +6,12 @@ APPEND_GLOB(SOURCES ${DIR}/*.cpp)
 # Executables
 #
 
-IF(${PACKAGE_NAME}_ENABLE_Epetra)
+IF(${PACKAGE_NAME}_ENABLE_Epetra AND ${PACKAGE_NAME}_ENABLE_ML)
```
Comment (lucbv) [inline packages/muelu/research/mmayr/composite_to_regions/src/CMakeLists.txt]: do we need to enable ML here? I thought Matthias' work only required MueLu?

**Comment by jhux2** [comment]
@bartlettroscoe How do I find the *exact* cmake configure line that the PR bot is using?  My local build does not show these failures.

**Comment by bartlettroscoe** [comment]
@jhux2 said:

> @bartlettroscoe How do I find the exact cmake configure line that the PR bot is using? My local build does not show these failures.

Likely easiest to reproduce this on a SNL COE RHEL 6 machine with the SEMS env with the ATDM configuration.  The general instructions for that are given at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#sems-rhel6-environment

But given this test was disabled in commit 06b1ca0ad6babdeb5b76c62a28d9c7270d795f52 you will need to either re-enable it by reverting that commit in your local branch or just configure setting `-DMueLu_UnitTestsTpetra_MPI_4_DISABLE=FALSE`.  So that would be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh gcc-opt-openmp

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_MueLu=ON \
  --DMueLu_UnitTestsTpetra_MPI_4_DISABLE=FALSE \
  $TRILINOS_DIR

$ make NP=16

$ ctest -j16
```

Let me know if that does not work.

**Comment by bartlettroscoe** [comment]
@jhux2, sorry the commit that disables this test in all ATDM builds was 5e52db03ff33acb5b9a0be7ba7507a8bb0de6e30 but the same cache var `-DMueLu_UnitTestsTpetra_MPI_4_DISABLE=FALSE` should allow it to be enabled.

**Comment by jhux2** [comment]
> @jhux2, sorry the commit that disables this test in all ATDM builds was 5e52db0 but the same cache var -DMueLu_UnitTestsTpetra_MPI_4_DISABLE=FALSE should allow it to be enabled.

@bartlettroscoe Thanks, I'll use the configure you suggested.

**Comment by jhux2** [comment]
@bartlettroscoe Is this message expected?
```
(~/.../trilinosBuilds/build-issue-3035) printenv TRILINOS_HOME
/home/jhu/software/Trilinos-github
(~/.../trilinosBuilds/build-issue-3035) source $TRILINOS_HOME/cmake/std/atdm/load-env.sh gcc-opt-openmp
Hostname 'typhon.ca.sandia.gov' matches known ATDM host 'sems-rhel6' and system 'rhel6'
ATDM_CONFIG_TRILNOS_DIR = /home/jhu/software/Trilinos-github
Setting default compiler and build options for JOB_NAME='gcc-opt-openmp'
***
*** ERROR: A compiler was not specified in 'gcc-opt-openmp'!
***
Using SEMS RHEL6 compiler stack GNU to build RELEASE code with Kokkos node type OPENMP
(~/.../trilinosBuilds/build-issue-3035) which gcc
/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/6.1.0/base/bin/gcc
```

**Comment by jhux2** [comment]
Replacing `gcc` with `gnu` works:

```
(develop>) (~/.../cmake/std) source $TRILINOS_HOME/cmake/std/atdm/load-env.sh gnu-opt-openmp
Hostname 'typhon.ca.sandia.gov' matches known ATDM host 'sems-rhel6' and system 'rhel6'
ATDM_CONFIG_TRILNOS_DIR = /home/jhu/software/Trilinos-github
Setting default compiler and build options for JOB_NAME='gnu-opt-openmp'
Using SEMS RHEL6 compiler stack GNU to build RELEASE code with Kokkos node type OPENMP
```

**Comment by jhux2** [comment]
@bartlettroscoe But it still results in using gcc 6.1.0, not gcc 4.8.4 as on the PR dashboard.  Anyway, I'll go with what I have in hand.

**Comment by bartlettroscoe** [comment]
@jhux2 said:

> Replacing gcc with gnu works:

Sorry, my bad.  We should improve that error message to give the valid list of compilers.

**Comment by jhux2** [comment]
Using the autotester cmake configure and SEMS setup, I cannot reproduce the problem.   However, I see a lot of other MueLu tests failing.  It's possible that something is different on my workstation (COE RHEL6), but I have no idea what.

**Comment by jhux2** [comment]
> Using the autotester cmake configure and SEMS setup, I cannot reproduce the problem. However, I see a lot of other MueLu tests failing. It's possible that something is different on my workstation (COE RHEL6), but I have no idea what.

For whatever reason, Isorropia was not enabled through the ATDM scripts -- MueLu has an optional dependency on it.  I've enabled it by hand and am rebuilding.

**Comment by jhux2** [comment]
Ok, now I can recreate the dashboard failure. @bartlettroscoe Isorropia wasn't enabled via the ATDM scripts, I had to do that by hand.  Isorropia is required to reproduce the failure.

**Pull request closed without merging**